lib/Twig/Environment.php

```diff
@@ -1067,11 +1079,11 @@ public function addGlobal($name, $value)
                 $this->globals = $this->initGlobals();
             }
 
-            /* This condition must be uncommented in Twig 2.0
             if (!array_key_exists($name, $this->globals)) {
-                throw new LogicException(sprintf('Unable to add global "%s" as the runtime or the extensions have already been initialized.', $name));
+                // The deprecation notice must be turned into the following exception in Twig 2.0
+                @trigger_error(sprintf('Registering global variable "%s" at runtime or when the extensions have already been initialized is deprecated.', $name), E_USER_DEPRECATED);
+                //throw new LogicException(sprintf('Unable to add global "%s" as the runtime or the extensions have already been initialized.', $name));
```

Comment:
I don't really see the point here, to use a deprecated error notice instead of an exception and then suggest to actually use the same exception (unless I read it wrong) in 2.0 ?